packages/hardhat-core/test/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -1008,6 +1031,18 @@ describe("Eth module", function () {
             1
           );
         });
+
+        it("Should return 1 for a pending block", async function () {
```

Comment:
This is not always the case. The method could return 2 if there were two transactions in the mempool. I would change the name of this test
```suggestion
        it("Should return the number of transactions in the 'pending' block", async function () {
```
and also update the name of the previous one.